src/query/plan/operator.cpp

```diff
@@ -683,6 +684,13 @@ bool Expand::ExpandCursor::Pull(Frame &frame, ExecutionContext &context) {
     // attempt to get a value from the incoming edges
     if (in_edges_ && *in_edges_it_ != in_edges_->end()) {
       auto edge = *(*in_edges_it_)++;
+      if (context.auth_checker &&
+          (!context.auth_checker->IsUserAuthorizedEdgeType(context.user, context.db_accessor, edge.EdgeType()) ||
```

Comment:
We are here checking both 2 vertices and an edge. Since the ScanAll operator is preceding Expand, we can definitely reduce the number of checks to 2 instead of 3.